### PR TITLE
fix: Report exception if filewatcher fails to start

### PR DESCRIPTION
**Pull request by Artur-**
Related to #17159

### CODE REVIEW

File: vaadin-dev-server/src/main/java/com/vaadin/base/devserver/FileWatcher.java [hunk 1]
```diff
@@ -61,7 +65,14 @@ public FileWatcher(SerializableConsumer<File> onChangeConsumer,
...
     public void start() {
-        watcher.watchAsync();
+        CompletableFuture<Void> future = watcher.watchAsync();
+        if (future.isCompletedExceptionally()) {
+            try {
+                future.get();
+            } catch (Exception e) {
+                getLogger().error("Error starting file watcher", e);
+            }
+        }
```
Comment (mcollovati) [inline vaadin-dev-server/src/main/java/com/vaadin/base/devserver/FileWatcher.java:75]: Can this be simplified using `CompletionStage.exceptionally()`?

```java
watcher.exceptionally(ex -> {
            getLogger().error("Error starting file watcher", ex);
            return null;
        });
```

File: vaadin-dev-server/src/main/java/com/vaadin/base/devserver/FileWatcher.java [hunk 2]
```diff
@@ -74,4 +85,8 @@ public void stop() throws IOException {
         watcher.close();
     }
 
+    private Logger getLogger() {
+        return LoggerFactory.getLogger(getClass());
+    }
+
```
Comment (mcollovati) [inline vaadin-dev-server/src/main/java/com/vaadin/base/devserver/FileWatcher.java:91]: Other classes in the same package define `getLogger` as a static method. Shall we adopt the same pattern here?

File: vaadin-dev-server/src/main/java/com/vaadin/base/devserver/FileWatcher.java (hunk 1 above)
Comment (Artur-) [reply vaadin-dev-server/src/main/java/com/vaadin/base/devserver/FileWatcher.java]: Yes, much better

File: vaadin-dev-server/src/main/java/com/vaadin/base/devserver/FileWatcher.java (hunk 2 above)
Comment (Artur-) [reply vaadin-dev-server/src/main/java/com/vaadin/base/devserver/FileWatcher.java]: Sure

**Review by mcollovati — APPROVED** [review]

**Pull request merged**